src/swtpm/swtpm_chardev.c

```diff
@@ -126,12 +126,17 @@ static void usage(FILE *file, const char *prgname, const char *iface)
     "                 : use the given character device\n"
     "-f|--fd <fd>     : use the given character device file descriptor\n"
     "-d|--daemon      : daemonize the TPM\n"
-    "--ctrl type=[unixio|tcp][,path=<path>][,port=<port>[,bindaddr=address[,ifname=ifname]]][,fd=<filedescriptor]\n"
+    "--ctrl type=[unixio|tcp][,path=<path>][,port=<port>[,bindaddr=address[,ifname=ifname]]][,fd=<filedescriptor|clientfd=<filedescriptor>]\n"
     "                 : TPM control channel using either UnixIO or TCP sockets;\n"
     "                   the path is only valid for Unixio channels; the port must\n"
     "                   be given in case the type is TCP; the TCP socket is bound\n"
     "                   to 127.0.0.1 by default and other bind addresses can be\n"
-    "                   given with the bindaddr parameter\n"
+    "                   given with the bindaddr parameter; if fd is provided,\n"
+    "                   it will be treated as a server socket and used for \n"
+    "                   accepting client connections; if clientfd is provided,\n"
+    "                   it will be treaded as client connection;\n"
+    "                   NOTE: fd and clientfd are mutually exclusive and only\n"
+    "                   valid for UinxIO channels\n"
```

Comment:
The overall idea of '--fd, --client' fd is to use swtpm as peer-2-peer server. That allows to fork from a process and communicate using socket pairs. The intention is to start swtpm inside qemu tpm backend and communicate over unix sockets, Here are the proposed qemu changes(your review comments are most appreciated): 

 https://github.com/avalluri/qemu-tpm/commit/3d05ba019453d1659cf1d2b5aaae5db7ade5cc6f#diff-a05dbc826a94c3e144c3ab1e12d8f364R394